### PR TITLE
Allow topic names for destroy topic

### PR DESCRIPTION
**Pull request by pdeziel**
This PR updates the `destroy_topic` method to accept topic names as well as IDs.

### CHECKLIST

- [x] _Is the commit message formatted correctly?_

- [x] _Do all of your functions and methods have docstrings?_
- [x] _Have you added/updated unit tests where appropriate?_
- [x] _Have you run the unit tests using `pytest`?_
- [x] _Is your code style correct (are you using PEP8, pyflakes)?_

### CODE REVIEW

File: pyensign/ensign.py [hunk 1]
```diff
@@ -423,17 +424,30 @@ async def destroy_topic(self, id: str) -> bool:
...
+        except CacheMissError:
+            # If the ID is not in the cache, look it up from the server
+            id = await self.topic_id(topic)
+
         _, state = await self.client.destroy_topic(id)
-        return state == topic_pb2.TopicState.DELETING
+        if state != TopicState.DELETING:
+            raise EnsignTopicDestroyError(
+                "failed to destroy topic, topic state is {}".format(state)
+            )
```
Comment (pdeziel) [inline pyensign/ensign.py:450]: Changed this to raise an exception rather than return a bool which is a bit more pythonic I think.

Comment (pdamodaran) [reply pyensign/ensign.py:450]: I like this change - it is more informative as well

File: pyensign/connection.py [hunk 2]
```diff
@@ -284,7 +285,7 @@ async def destroy_topic(self, id):
             id=id, operation=topic_pb2.TopicMod.Operation.DESTROY
         )
         rep = await self.stub.DeleteTopic(params)
-        return rep.id, rep.state
+        return rep.id, TopicState.convert(rep.state)
```
Comment (pdamodaran) [inline pyensign/connection.py:288]: nice!

**Review by pdamodaran — APPROVED** [review]
looks good @pdeziel!

**Pull request merged**